tokio-postgres/tests/test/main.rs

```diff
@@ -347,7 +347,250 @@ async fn custom_range() {
+        let client = connect("user=postgres").await;
+
+        match client.simple_query("DROP TABLE sbtest1").await {
+            Ok(_) => {
+                println!("Query executed successfully");
+                // Your code here when the query succeeds
+            }
+            Err(e) => {
+                eprintln!("Error executing query: {}", e);
+                // Your code here when the query fails
+            }
+        }
+
+        match client.simple_query("DROP TABLE sbtest2").await {
+            Ok(_) => {
+                println!("Query executed successfully");
+                // Your code here when the query succeeds
+            }
+            Err(e) => {
+                eprintln!("Error executing query: {}", e);
+                // Your code here when the query fails
+            }
+        }
+
+        match client
+            .simple_query("CREATE TABLE sbtest1 (id INTEGER, k INTEGER);")
+            .await
+        {
+            Ok(_) => {
+                println!("Query executed successfully");
+                // Your code here when the query succeeds
+            }
+            Err(e) => {
+                eprintln!("Error executing query: {}", e);
+                // Your code here when the query fails
+            }
+        }
+
+        let messages = client
+            .simple_query(
+                "INSERT INTO sbtest1  VALUES (1, 2);
+            INSERT INTO sbtest2  VALUES (1, 2);
+            SELECT * FROM sbtest1 ORDER BY id;
+            SELECT k FROM sbtest1 WHERE id = 999;
+            BEGIN;
+            UPDATE sbtest1 SET k=id;
+            UPDATE sbtest2 SET k=id;
+            END;",
+            )
+            .await
+            .unwrap();
+
+        match messages[0] {
+            SimpleQueryMessage::CommandComplete(CommandCompleteContents { rows: 1, .. }) => {}
+            _ => panic!("unexpected message"),
+        }
+        match messages[1] {
+            SimpleQueryMessage::CommandComplete(CommandCompleteContents { rows: 1, .. }) => {}
+            _ => panic!("unexpected message"),
+        }
+        match &messages[2] {
+            SimpleQueryMessage::Row(row) => {
+                assert_eq!(row.columns().get(0).map(|c| c.name()), Some("id"));
+                assert_eq!(row.columns().get(1).map(|c| c.name()), Some("k"));
+                assert_eq!(row.get(0), Some("1"));
+                assert_eq!(row.get(1), Some("2"));
+            }
+            _ => panic!("unexpected message"),
+        }
+        match &messages[3] {
+            SimpleQueryMessage::CommandComplete(CommandCompleteContents { fields: f, .. }) => {
+                if f.is_some() {
+                    panic!("Unexpected data found");
+                }
+            }
+            _ => panic!("unexpected message"),
+        }
+        match &messages[4] {
+            SimpleQueryMessage::CommandComplete(CommandCompleteContents { fields: f, .. }) => {
+                if let Some(field_vec) = &f {
```

Comment:
I think you can match with `SimpleQueryMessage::CommandComplete(CommandCompleteContests { fields: Some(f), ... })` instead of checking whether fields is_some() in the body.